src/apply.c

```diff
@@ -24,7 +24,7 @@
 #include "reader.h"
 
 #define apply_err(...) \
-	( giterr_set(GITERR_PATCH, __VA_ARGS__), -1 )
+	( giterr_set(GITERR_PATCH, __VA_ARGS__), GIT_EAPPLYFAIL )
```

Comment:
That's debatable, but I'd rather add a `GITERR_APPLY` error class (so you can mix-and-match with error codes, eg `GIT_EINVALID` in this case) instead of adding a new error code ?

Reply:
I'm happy to change the error class here if you think that it would be useful, but I think that having an actual `GIT_EAPPLYFAIL` error code is useful.  I don't think that there's any other error code that suggests that patch application was impossible.  `GIT_ECONFLICT` is probably the _closest_ but it's not right.